### PR TITLE
Fixes #5684: cancel repository discovery is now dynflowed

### PR DESCRIPTION
**Pull request by xprazak2**
Repository discovery can now be cancelled

### CODE REVIEW

**Comment by ehelms** [comment]
It's not obvious to me why the RepoDiscovery class needed to be moved into the dynflow action for this.

**Comment by xprazak2** [comment]
I need to periodically suspend dynflow action during disovery because it checks for cancel event when it wakes up. Discovery progress is noted in output hash, so if there is no cancel, discovery continues where it left off before it was suspended. I am not sure if it was completely necessary to move RepoDiscovery code into Discover action class, but it seemed the most straightforward approach to me.

**Comment by ehelms** [comment]
What strikes me about that change is we have a nice little object that
knows of to do one thing - repo discovery. Then we also have a dynflow task
that needs to use that object and wrap some orchestration around it.
Combining the two just seems to muddy the waters between the two concerns.

**Comment by xprazak2** [comment]
> Then we also have a dynflow task that needs to use that object and wrap some orchestration around it.
I did not look at it that way because I needed to break the whole discovery into sequence of steps. Should I rewrite it and preserve RepoDiscovery class?

**Comment by ehelms** [comment]
If possible, I would prefer that to keep separate the concerns and make it more readable when looking at the dynflow task itself.

**Comment by xprazak2** [comment]
OK

File: app/controllers/katello/api/v2/organizations_controller.rb [hunk 1]
```diff
@@ -92,8 +92,8 @@ def repo_discover
     param :label, String, :desc => N_("Organization label")
     param :url, String, :desc => N_("base url to perform repo discovery on")
     def cancel_repo_discover
-      # TODO: implement task canceling
-      render :json => { message: "not implemented" }
+      @organization.cancel_repo_discovery
+      render :nothing => true
```
Comment (ehelms) [inline app/controllers/katello/api/v2/organizations_controller.rb]: I feel like the user should be getting a task back or some information to indicate the result of requested action and whether it was successful or not.

**Comment by xprazak2** [comment]
User gets now a task back

**Comment by ehelms** [comment]
@xprazak2 apologies for the delay in getting back to this, going to test/review again today

**Comment by ehelms** [comment]
@xprazak2 looks good and worked well for me, thanks! ACK

**Pull request merged**